src/wieldmesh.cpp

```diff
@@ -531,30 +558,30 @@ void getItemMesh(Client *client, const ItemStack &item, ItemMesh *result)
 					break;
 				}
 				default: {
-					MeshMakeData mesh_make_data(client, false);
-					MapNode mesh_make_node(id, 255, 0);
-					mesh_make_data.fillSingleNode(&mesh_make_node);
-					MapBlockMesh mapblock_mesh(&mesh_make_data, v3s16(0, 0, 0));
-					mesh = cloneMesh(mapblock_mesh.getMesh());
-					translateMesh(mesh, v3f(-BS, -BS, -BS));
+					MeshMakeData mesh_make_data(client, false, false);
```

Comment:
Could you please move this and the changed part above into a new function to
1) reduce the amount of indents
2) less code duplication?
I hope a few more checks are acceptable in terms of speed.

Reply:
Am I right that textures after 0th are not ever used (even if present) if shaders are disabled?

Reply:
According to https://github.com/minetest/minetest/blob/20a85d7/doc/lua_api.txt#L968 there can be multiple textures (tiles) for a mesh.

Reply:
I mean textures after 0th in a single material.